cli/server.go

```diff
@@ -2167,6 +2132,90 @@ func ConfigureHTTPServers(inv *clibase.Invocation, cfg *codersdk.DeploymentValue
 	return httpServers, nil
 }
 
+func createDERPMap(ctx context.Context, logger slog.Logger, vals *codersdk.DeploymentValues) (*tailcfg.DERPMap, error) {
+	_, accessURLPortRaw, _ := net.SplitHostPort(vals.AccessURL.Host)
+	if accessURLPortRaw == "" {
+		accessURLPortRaw = "80"
+		if vals.AccessURL.Scheme == "https" {
+			accessURLPortRaw = "443"
+		}
+	}
+
+	accessURLPort, err := strconv.Atoi(accessURLPortRaw)
+	if err != nil {
+		return nil, xerrors.Errorf("parse access URL port: %w", err)
+	}
+
+	defaultRegion := &tailcfg.DERPRegion{
+		EmbeddedRelay: true,
+		RegionID:      int(vals.DERP.Server.RegionID.Value()),
+		RegionCode:    vals.DERP.Server.RegionCode.String(),
+		RegionName:    vals.DERP.Server.RegionName.String(),
+		Nodes: []*tailcfg.DERPNode{{
+			Name:      fmt.Sprintf("%db", vals.DERP.Server.RegionID),
+			RegionID:  int(vals.DERP.Server.RegionID.Value()),
+			HostName:  vals.AccessURL.Value().Hostname(),
+			DERPPort:  accessURLPort,
+			STUNPort:  -1,
+			ForceHTTP: vals.AccessURL.Scheme == "http",
+		}},
+	}
+	if !vals.DERP.Server.Enable {
+		defaultRegion = nil
+	}
```

Comment:
This logic feels a bit backwards to me. I think it would be clearer as 
```go
var defaultRegion *tailcfg.DERPRegion
if vals.DERP.Server.Enable { defaultRegion = ... }
```